sync/src/blocks.rs

```diff
@@ -366,23 +370,26 @@ impl BlockCollection {
 			let receipts = UntrustedRlp::new(&r);
 			ordered_trie_root(receipts.iter().map(|r| r.as_raw().to_vec())) //TODO: get rid of vectors here
 		};
-		match self.receipt_ids.get(&receipt_root).cloned() {
-			Some(h) => {
-				self.receipt_ids.remove(&receipt_root);
+		match self.receipt_ids.entry(receipt_root) {
+			Entry::Occupied(mut entry) => {
+				let h = entry.get_mut().pop().expect("Empty vectors are not allowed in insert_receipt; qed");
```

Comment:
actually, if the receipts roots are all the same here then we should be able to drain all block hashes in the entry as soon as we get the correct set.